File: programmers/python 3/49993.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/49993

def solution(skill, skill_trees):
    skill_trees = [
        ''.join(filter(lambda x: x in skill, tree)) 
        for tree in skill_trees
    ]
    return sum(skill.startswith(tree) for tree in skill_trees)

File: programmers/swift/49993.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/49993

func solution(_ skill: String, _ skillTrees: [String]) -> Int {
    skillTrees
        .map { skill.starts(with: $0.filter { skill.contains($0) }) ? 1 : 0 }
        .reduce(0, +)
}